compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -34,4 +40,125 @@ luci::CircleNode *CloneNodeLet<CN::OPQR>::visit(const luci::CircleReshape *node)
   return cloned;
 }
 
+namespace sinf
+{
+
+/**
+ * @note  CircleReshape always has two inputs: tensor and shape.
+ *        The shape input can be CircleConst, CircleOutputDummy, or CircleNode.
+ *        - If the shape input is CircleConst, the shape is inferred from the constant.
+ *        - If the shape input is CircleOutputDummy, the shape is inferred from
+ *          the attribute or the node itself.
+ *        - If the shape input is CircleNode, the shape is not inferred.
+ */
+loco::TensorShape Algorithm::visit(const luci::CircleReshape *node)
+{
+  const loco::DataType S32 = loco::DataType::S32;
+
+  // CircleReshape node must have reshape/shape
+  if (node->shape() == nullptr)
+  {
+    INTERNAL_EXN("2nd input shape() should not be nullptr");
+  }
+
+  bool should_infer = true;
+  loco::TensorShape output_shape;
+  {
+    // Check if reshape/shape is CircleConst
+    auto const_input = dynamic_cast<luci::CircleConst *>(node->shape());
+    if (const_input != nullptr)
+    {
+      output_shape.rank(const_input->size<S32>());
+
+      for (uint32_t axis = 0; axis < output_shape.rank(); ++axis)
+      {
+        output_shape.dim(axis) = const_input->at<S32>(axis);
```

Comment:
If `const_input` has -1 value, the dimension value becomes 0xFFFFFFFF and `known() == true`.
When this happens, `output_shape.dim(axis).unset()` would be correct.

Reply:
I missed it...!

Thanks for pointing really important thing out :)

I'll make the change.